src/admin/components/SearchForm.js

```diff
@@ -44,24 +44,25 @@ export class AdminSearchFormBase extends React.Component {
   }
 
   handleSearch = () => {
-    this.goToSearch(this.refs.query.value);
+    this.goToSearch(this.query.value);
   }
 
   render() {
     const { pathname, query } = this.props;
     return (
-      <form ref="form" className="search-form" method="GET"
-            action={pathname} onKeyDown={this.handleKeyDown}
+      <form method="GET" action={pathname} className="AdminSearchForm-form"
+            ref={(ref) => { this.form = ref; }} onKeyDown={this.handleKeyDown}
             onSubmit={(e) => e.preventDefault()}>
         <label className="visually-hidden" htmlFor="q">{_('Search')}</label>
-        <input ref="query" type="search" name="q" placeholder={_('Search')}
-               defaultValue={query} />
+        <input ref={(ref) => { this.query = ref; }} type="search" name="q"
```

Comment:
Oh, fair enough. I don't know how to enforce that with the linter, but if we could I say yes.

Otherwise it's on us to enforce, which I guess works but is annoying.